File: skills/vuln-discovery/evals/fixtures/javascript/cmdi_vulnerable.js
/**
 * Vulnerable: Command injection via child_process.exec with user input.
 */
const express = require("express");
const { exec } = require("child_process");
const app = express();

app.get("/ping", (req, res) => {
  const host = req.query.host;
  exec(`ping -c 1 ${host}`, (error, stdout, stderr) => {
    res.send(`<pre>${stdout}</pre>`);
  });
});

app.get("/lookup", (req, res) => {
  const domain = req.query.domain;
  exec("nslookup " + domain, (error, stdout) => {
    res.json({ result: stdout });
  });
});

File: skills/vuln-discovery/evals/fixtures/python/cmdi_vulnerable.py
"""Vulnerable: OS command injection via subprocess with shell=True."""
import subprocess

def ping_host(host):
    result = subprocess.run(f"ping -c 1 {host}", shell=True, capture_output=True, text=True)
    return result.stdout

def list_dir(path):
    import os
    os.system("ls -la " + path)
